worldwind/src/commonMain/kotlin/earth/worldwind/shape/AbstractShape.kt

```diff
@@ -44,14 +45,36 @@ abstract class AbstractShape(override var attributes: ShapeAttributes): Abstract
     override var highlightAttributes: ShapeAttributes? = null
     override var isHighlighted = false
     var maximumIntermediatePoints = 10
-    protected lateinit var activeAttributes: ShapeAttributes
-    protected var isSurfaceShape = false
+    lateinit var activeAttributes: ShapeAttributes
+        protected set
+    var isSurfaceShape = false
+        protected set
     protected var lastGlobeState: Globe.State? = null
     protected var pickedObjectId = 0
     protected val pickColor = Color()
     protected val boundingSector = Sector()
     protected val boundingBox = BoundingBox()
     private val scratchPoint = Vec3()
+    private var activeAttributesHash = 0
+    open var allowBatching = false
+    var forceRecreateBatch = false
+    var lastRequestedFrameIndex = 0L
+
+    open fun addToBatch(rc : RenderContext) : Boolean {
+        return rc.currentLayer is RenderableLayer && allowBatching && !isHighlighted
```

Comment:
In case of Graticule it will not b a renderable layer. Can we remove this cast check?

Reply:
Added TODO comment, better to address that in another PR with more testing